packages/core/src/editor/model/Editor.ts

```diff
@@ -1136,4 +1137,69 @@ export default class EditorModel extends Model {
       el[varName][name] = value;
     }
   }
+
+  private sendTelemetryData() {
+    const hostName = window.location.hostname;
+
+    // @ts-ignore
+    const enableDevTelemetry = __ENABLE_TELEMETRY_LOCALHOST__;
+
+    if (!enableDevTelemetry && (hostName === 'localhost' || hostName.includes('localhost'))) {
+      // Don't send telemetry data for localhost
+      return;
+    }
+
+    const sessionKeyPrefix = 'gjs_telemetry_sent_';
+
+    // @ts-ignore
+    const version = __GJS_VERSION__;
+    const sessionKey = `${sessionKeyPrefix}${version}`;
+
+    if (sessionStorage.getItem(sessionKey)) {
+      console.log(`Telemetry already sent for version ${version} this session`);
+      return;
+    }
+
+    // @ts-ignore
+    const url = __STUDIO_URL__;
+    const path = '/api/gjs/telemetry/collect';
+
+    (async () => {
+      const response = await fetch(`${url}${path}`, {
```

Comment:
let's make `sendTelemetryData` async so you can avoid IIFE here?

Reply:
The IFFE is so we don't block sending telemetry data, the function will be added to the queue and resolved when appropriate. Also its an IFFE because I am not sure if backbone event listeners can be async.